### DISCUSSION TITLE
/using-the-office-365-cli-in-your-vsts-ci-cd-pipelines-for-sharepoint-framework-solutions/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#293] [Comments]
Using the Office 365 CLI in your VSTS CI/CD pipelines for SharePoint Framework solutions
        
[https://www.eliostruyf.com/using-the-office-365-cli-in-your-vsts-ci-cd-pipelines-for-sharepoint-framework-solutions/](https://www.eliostruyf.com/using-the-office-365-cli-in-your-vsts-ci-cd-pipelines-for-sharepoint-framework-solutions/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Devaraj Gowda Mudigere on 2018-04-19 12:35
```

Hi Elio, I have the set of tasks created, my build works without issues locally with all commands.  
When i configure the same in VSTS it works fine without using --ship.

If i use --ship anywhere in build/bundle then it fails at the end, here still the build is succesfull, but terminates ship.  
here is the log, it would be very helpful if this issue fixed  
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*  
Finished subtask 'copy-assets' after 2.18 s  
Starting subtask 'write-manifests'...  
Finished subtask 'write-manifests' after 399 ms  
Finished 'bundle' after 36 s  
\==================\[ Finished \]==================  
Warning - Setting build task  
Project version: 0.0.1  
Build tools version: 3.2.7  
Node version: v8.11.1  
Total duration: 40 s  
Task warnings: 1  
Gulp failed with error: C:\\NPM\\Modules\\gulp.cmd failed with return code: 1

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-04-19 12:39
```

It is because of the warning. The VSTS task sees the warning as an error, so it will stop the process. So check in your output what the warning is, and fix it. You can also specify in the task itself to just continue on errors, or you can add suppression for the warning in the gulpfile.js file like this:

build.addSuppression(\`Warning - \[sass\] The local CSS class 'ms-Grid' is not camelCase and will not be type-safe.\`);

**Reply by comment-migrator** [reply]
```header
Originally posted by Devaraj Gowda Mudigere on 2018-04-19 14:30
```

Thank you Elio, That works !.  
All steps are successful, but i am not seeing the package deployed in app catalog. What could be the reason ?

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-04-19 14:47
```

1\. Is the package added?  
2\. If it is added, it might went wrong it the deployment process. In that case check if you provided the right title + flags.  
3\. If it is not added, check if you can run it from your own machine with the same credentials?

**Reply by comment-migrator** [reply]
```header
Originally posted by Devaraj Gowda Mudigere on 2018-04-19 14:54
```

Its giving me so many problems from morning..

More packages we add the more problems are!

![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/7c078a59a11fe128ec6abf592a7d7342d590959a108fa88dc04d6e49919023bd.png)

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-04-19 15:02
```

Check the logs for the actual error. You could try the "Hosted" agent instead of the "Hosted VS2017" one. I always use the "hosted" one.

**Reply by comment-migrator** [reply]
```header
Originally posted by Devaraj Gowda Mudigere on 2018-04-19 15:38
```

Yeah! Tried hosted first, then tried vs2017

**Reply by comment-migrator** [reply]
```header
Originally posted by JH on 2018-04-20 12:38
```

Elio. We too are recently seeing the same issue (Gulp failed with error: C:\\npm\\prefix\\gulp.cmd). This is with deployments that have previously worked ok end-to-end.

When I check the logs, I can see a 404 response due to 'undefined' in path name when getting the app catalog and attempting to upload the sppkg (/undefined/\_api/web/GetFolderByServerRelativeUrl(...).

Does this suggest anything obvious? I am unsure as to where I should start debugging.

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-04-20 12:44
```

Are you also using the Office 365 CLI for adding the package? If yes, does it work on your own machine?

**Reply by comment-migrator** [reply]
```header
Originally posted by JH on 2018-04-20 12:52
```

Thanks for the quick response. We are not using CLI, but Gulp with the NPM package.

**Reply by comment-migrator** [reply]
```header
Originally posted by Devaraj Gowda Mudigere on 2018-04-20 17:01
```

Elio, it's working for me, thanks for your suggestion.

John - Check logs line by line, fix warnings and erros if there are any.. Warnings are not marked with any diff color, so that will be the case, keep an eye!

**Reply by comment-migrator** [reply]
```header
Originally posted on 2018-04-26 08:58
```

John are the correct environment variables provided? As I see an undefined in you URL, I suspect the tenant or site URL wasn't provided.

**Reply by comment-migrator** [reply]
```header
Originally posted by JH on 2018-04-26 09:30
```

I think you are right. We are in the process of making some fundamental changes to our CI/CD process, part of which is to ensure correct variables and make things work more closely to your example above, including CLI. Tests using a command prompt show it to work reliably so I expect that this will resolve the issue.

**Comment by comment-migrator** [comment]
```header
Originally posted by Raymond Little on 2018-04-19 15:12
```

I've been using these build process in our company and while we're only starting out I can see the potential. We're are currently on the basic package for VSTS and as such only get 240 min each month of build time and have just gone over that. I was wondering do you actively use this in practice and have you found you need to upgrade to Microsoft-hosted CI/CD?

Cheers

**Reply by comment-migrator** [reply]
```header
Originally posted on 2019-02-13 08:29
```

Yes, we are very actively using this. We do multiple builds and releases per day. So we really relying on it and are paying for the Microsoft-hosted CI/CD plan, as we need more than 240min per month.

**Comment by comment-migrator** [comment]
```header
Originally posted by Ravi on 2019-01-16 16:29
```

I got below error when connecting SPO with O365 CLI

The user or administrator has not consented to use the application with ID 'GUID' named 'PnP Office 365 Management Shell'. Send an interactive authorization request for this user and resource.

**Reply by comment-migrator** [reply]
```header
Originally posted by Ravi on 2019-01-16 17:28
```

I got it working. Had to log in once from my machine with O365 CLI. From there I consent to "PnP Office 365 Management Shell" and it worked our.

Below link help me to understand the process

[https://pnp.github.io/offic...](https://pnp.github.io/office365-cli/user-guide/connecting-office-365/ "https://pnp.github.io/office365-cli/user-guide/connecting-office-365/")

**Reply by comment-migrator** [reply]
```header
Originally posted on 2019-02-13 08:25
```

The very first time you use the Office 365 CLI, you will have to indeed give your consent in order to let the app do things in your name. Once you gave your consent, this will not appear again.

**Comment by comment-migrator** [comment]
```header
Originally posted by Ravi on 2019-08-20 21:34
```

Hi Elio,

This was working and all of a sudden my pipeline stopped due to below error.

I also added --debug -- verbose to he command but couldn't get enough details.

**Comment by comment-migrator** [comment]
```header
Originally posted by Ravi on 2019-08-20 21:35
```

![https://uploads.disquscdn.c...](https://uploads.disquscdn.com/images/660885a63006596487c1bb19f33cb9f84c3238478e2ff163e3f4393f64f144aa.png)

**Reply by comment-migrator** [reply]
```header
Originally posted on 2019-08-24 21:29
```

This seems to be a sub-dependency issue. Try to use the lastest (or an older) CLI version.

Do you also have the issue if you install the same CLI version on your computer?

One last thing, which node version are you using on Azure DevOps? Be sure to use at least v8.

**Comment by Debabrata18992** [comment]
I tried to login using  spo connect https://estruyfdev2.sharepoint.com/sites/catalog --authType password --userName $(username) --password $(password) but it shows invalid command

**Reply by estruyf** [reply]
@Debabrata18992 it is an old post. These days, you need to use `m365 login` instead of `spo connect`. Check out the CLI docs for more information: https://pnp.github.io/cli-microsoft365/
